test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
> wait.Poll propagates the error if your function returns one

yes it will propagate the error if you return one, but it will also immediately stop retrying.

you cannot tell it to retry N times as long as your function returns an error, and at the end of that, if you're stil returning an error, propagate that error.